python/tank/platform/engine.py

```diff
@@ -979,8 +984,11 @@ def _do_invoke(self):
                         Execute function and return result
                         """
                         self._res = self._fn()
-                        
-                return Invoker()
+
+                # Make sure that the invoker is for the main thread only.
+                invoker = Invoker()
+                invoker.moveToThread(QtCore.QThread.currentThread())
```

Comment:
Is this a bug we are fixing? Just curious! ( @alandann may also want to check this )